read_input_parser_cvc.l

```diff
@@ -642,43 +638,109 @@ static inline double fltlist_next_token(int * const list_end){
+<LOOPEXTRACTOUTQSQ>{FLT} {
+  cvc::g_LoopExtract_OutQSq = atof(yytext);
+  if(cvc::myverbose!=0) printf("# [read_input_parser] filtering momenta upto %f\n", cvc::g_LoopExtract_OutQSq);
 }
-<SPINPROJECTIONLOOPACCUMULATE>{FILENAME} {
+<LOOPEXTRACTNSTOCHACCUMULATED>{FILENAME} {
   if (strcmp(yytext,"yes")==0){
-    cvc::g_spinprojection_loopaccumulate = 1;
-    if(cvc::myverbose!=0) printf("# [read_input_parser] cvc::g_spinprojection_loopaccumulate is on\n");
+    cvc::g_LoopExtract_NstochAccumulated = 1;
+    if(cvc::myverbose!=0) printf("# [read_input_parser] cvc::g_LoopExtract_NstochAccumulated is on\n");
   }
   else if (strcmp(yytext,"no")==0){
-    cvc::g_spinprojection_loopaccumulate = 0;
-    if(cvc::myverbose!=0) printf("# [read_input_parser] cvc::g_spinprojection_loopaccumulate is off\n");
+    cvc::g_LoopExtract_NstochAccumulated = 0;
+    if(cvc::myverbose!=0) printf("# [read_input_parser] cvc::g_LoopExtract_NstochAccumulated is off\n");
   }
 }
-<SPINPROJECTIONSPINTRACE>{FILENAME} {
+<LOOPEXTRACTASCIIOUTPUT>{FILENAME} {
   if (strcmp(yytext,"yes")==0){
-    cvc::g_spinprojection_spintrace = 1;
-    if(cvc::myverbose!=0) printf("# [read_input_parser] cvc::g_spintrace is on\n");
+    cvc::g_LoopExtract_ASCII_Output = 1;
+    if(cvc::myverbose!=0) printf("# [read_input_parser] cvc::g_LoopExtract_ASCII_Output is on\n");
   }
   else if (strcmp(yytext,"no")==0){
-    cvc::g_spinprojection_spintrace = 0;
-    if(cvc::myverbose!=0) printf("# [read_input_parser] cvc::g_spintrace is off\n");
+    cvc::g_LoopExtract_ASCII_Output = 0;
+    if(cvc::myverbose!=0) printf("# [read_input_parser] cvc::g_LoopExtract_ASCII_Output is off\n");
   }
 }
-<SPINPROJECTIONGAMMABASIS>ukqcd {
-  cvc::g_spinprojection_gamma_basis = cvc::UKQCD;
+<LOOPEXTRACTSPINTRACE>{FILENAME} {
+  if (strcmp(yytext,"yes")==0){
+    cvc::g_LoopExtract_SpinTrace = 1;
+    if(cvc::myverbose!=0) printf("# [read_input_parser] cvc::g_LoopExtractSpinTrace is on\n");
+  }
+  else if (strcmp(yytext,"no")==0){
+    cvc::g_LoopExtract_SpinTrace = 0;
+    if(cvc::myverbose!=0) printf("# [read_input_parser] cvc::g_LoopExtractSpinTrace is off\n");
+  }
+}
+<LOOPEXTRACTSPINPROJECT>{FILENAME} {
+  if (strcmp(yytext,"yes")==0){
+    cvc::g_LoopExtract_SpinProject = 1;
+    if(cvc::myverbose!=0) printf("# [read_input_parser] cvc::g_LoopExtract_SpinProject is on\n");
+  }
+  else if (strcmp(yytext,"no")==0){
+    cvc::g_LoopExtract_SpinProject = 0;
+    if(cvc::myverbose!=0) printf("# [read_input_parser] cvc::g_LoopExtract_SpinProject is off\n");
+  }
+}
+<LOOPEXTRACTLEGACYTRAJ>{FILENAME} {
+  if (strcmp(yytext,"yes")==0){
+    cvc::g_LoopExtract_LegacyTraj = 1;
+    if(cvc::myverbose!=0) printf("# [read_input_parser] cvc::g_LoopExtract_LegacyTraj is on\n");
+  }
+  else if (strcmp(yytext,"no")==0){
+    cvc::g_LoopExtract_LegacyTraj = 0;
+    if(cvc::myverbose!=0) printf("# [read_input_parser] cvc::g_LoopExtract_LegacyTraj is off\n");
+  }
+}
+<LOOPEXTRACTSPINPROJECTGAMMASTRUCTURE>{CORRCONFIG} {
```

Comment:
how do I specify which gamma structures?

Reply:
you have to set LoopExtract_GammaStructure = 0,1,2,3,4, I will test this